diff --git a/EMailSender.cpp b/EMailSender.cpp
index 64e2907..2568f9d 100644
--- a/EMailSender.cpp
+++ b/EMailSender.cpp
@@ -457,6 +457,9 @@ EMailSender::Response EMailSender::send(const char* to[], byte sizeOfTo,  byte s
 	#endif
 #endif
 
+#ifndef PJW_ORIG
+uint8_t tBuf[2048];
+#endif
 EMailSender::Response EMailSender::send(const char* to[], byte sizeOfTo,  byte sizeOfCc,byte sizeOfCCn, EMailMessage &email, Attachments attachments)
 {
 #ifdef SSLCLIENT_WRAPPER
@@ -734,9 +737,16 @@ EMailSender::Response EMailSender::send(const char* to[], byte sizeOfTo,  byte s
 //  client.println(F("Mime-Version: 1.0"));
 
   client.println(F("MIME-Version: 1.0"));
+
+#ifdef PJW_ORIG
   client.println(F("Content-Type: Multipart/mixed; boundary=frontier"));
 
   client.println(F("--frontier"));
+#else
+  client.println(F("Content-Type: Multipart/mixed; boundary=frontier"));
+  client.println();
+  client.println(F("--frontier"));
+#endif
 
     client.print(F("Content-Type: "));
     client.print(email.mime);
@@ -771,8 +781,9 @@ EMailSender::Response EMailSender::send(const char* to[], byte sizeOfTo,  byte s
 	  DEBUG_PRINT(F("Array: "));
 //	  for (int i = 0; i<(sizeof(attachs) / sizeof(attachs[0])); i++){
 	  for (int i = 0; i<attachments.number; i++){
+#ifdef PJW_ORIG
 		  uint8_t tBuf[64];
-
+#endif
 		  if (attachments.fileDescriptor[i].url.length()==0){
 			  EMailSender::Response response;
 			  response.code = F("400");
@@ -884,7 +895,11 @@ EMailSender::Response EMailSender::send(const char* to[], byte sizeOfTo,  byte s
 						  encode(&myFile, &client);
 					  }else{
 						while(myFile.available()) {
+#ifndef PJW_ORIG
+						clientCount = myFile.read(tBuf,2048);
+#else
 						clientCount = myFile.read(tBuf,64);
+#endif
 						DEBUG_PRINTLN(clientCount);
 						  client.write((byte*)tBuf,clientCount);
 						}
@@ -951,7 +966,11 @@ EMailSender::Response EMailSender::send(const char* to[], byte sizeOfTo,  byte s
 				  }else{
 					  DEBUG_PRINTLN(F("NORMAL"));
 					while(myFile.available()) {
+#ifndef PJW_ORIG
+						clientCount = myFile.read(tBuf,2048);
+#else
 						clientCount = myFile.read(tBuf,64);
+#endif
 						client.write((byte*)tBuf,clientCount);
 					}
 				  }
@@ -984,7 +1003,10 @@ EMailSender::Response EMailSender::send(const char* to[], byte sizeOfTo,  byte s
 
 	  } // Close attachment cycle
 	  client.println();
+#ifdef PJW_ORIG
 	  client.println(F("--frontier--"));
+#endif
+
 #ifdef STORAGE_EXTERNAL_ENABLED
 	  #ifdef OPEN_CLOSE_SD
 		  if (sdActive){
@@ -1013,6 +1035,10 @@ EMailSender::Response EMailSender::send(const char* to[], byte sizeOfTo,  byte s
 #endif
 	  } // Close attachement enable
 
+#ifndef PJW_ORIG
+	  client.println(F("--frontier--"));
+#endif
+
 #endif
   DEBUG_PRINTLN(F("Message end"));
   client.println(F("."));
