--- C:\Users\trial\Desktop\ModbusMaster-2.0.0\src\ModbusMaster.cpp
+++ C:\Users\trial\Documents\Arduino\libraries\ModbusMaster\src\ModbusMaster.cpp
@@ -70,6 +70,11 @@
   pinMode(__MODBUSMASTER_DEBUG_PIN_A__, OUTPUT);
   pinMode(__MODBUSMASTER_DEBUG_PIN_B__, OUTPUT);
 #endif
+}
+
+void ModbusMaster::setSlaveId(uint8_t slave)
+{
+   _u8MBSlave = slave;
 }
 
 
--- C:\Users\trial\Desktop\ModbusMaster-2.0.0\src\ModbusMaster.h
+++ C:\Users\trial\Documents\Arduino\libraries\ModbusMaster\src\ModbusMaster.h
@@ -72,6 +72,8 @@
     ModbusMaster();
    
     void begin(uint8_t, Stream &serial);
+    void setSlaveId(uint8_t slave);
+
     void idle(void (*)());
     void preTransmission(void (*)());
     void postTransmission(void (*)());
